### PR TITLE
add go install as installation option

### PR DESCRIPTION
**Pull request by sombriks**
use go install results in a functional dbmate installation, as far as my tests goes.

please add this as installation option in the documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,14 @@ Install using [Scoop](https://scoop.sh)
 $ scoop install dbmate
 ```
 
+**GO Install**
+
+Remember to add `~/go/bin` into your `$PATH` if you decide to install dbmate this way.
```
Comment (dossy) [inline README.md:94]: `~/go/bin` is not correct in all cases.  A more generally correct recommendation would be `$(go env GOPATH)/bin`.

**Comment by amacneil** [comment]
I don't recommend this install option:

* No guarantee that go bin is in your $PATH
* Installing "latest commit on main" is never a good idea, you want a released version
* if you happen to run this inside an existing module directory you get

```sh
$ go install github.com/amacneil/dbmate/v2
no required module provides package github.com/amacneil/dbmate/v2; to add it:
	go get github.com/amacneil/dbmate/v2
```

Running the proposed `go get` command adds a ton of unrelated dependencies to your project.

Generally I don't see any reason you would want to build dbmate from source unless you are hacking on it or using as a library. Those that are doing so should already know the appropriate commands to run.

**Comment by sombriks** [comment]
> I don't recommend this install option:
> 
> * No guarantee that go bin is in your $PATH
> * Installing "latest commit on main" is never a good idea, you want a released version
> * if you happen to run this inside an existing module directory you get
> 
> ```shell
> $ go install github.com/amacneil/dbmate/v2
> no required module provides package github.com/amacneil/dbmate/v2; to add it:
> 	go get github.com/amacneil/dbmate/v2
> ```
> 
> Running the proposed `go get` command adds a ton of unrelated dependencies to your project.
> 
> Generally I don't see any reason you would want to build dbmate from source unless you are hacking on it or using as a library. Those that are doing so should already know the appropriate commands to run.

i had no idea it was intentional and understand the reasons. feel free to close this PR.

**Comment by amacneil** [comment]
Also I should mention: Since we import sqlite via cgo, you will need a C toolchain installed to compile that. So ya, I highly recommend sticking to prebuilt binaries.

**Pull request closed without merging**